scripts/config.pl

```diff
@@ -182,17 +214,19 @@
 
 my $done;
 for my $line (@config_lines) {
-    if ($action eq "full" || $action eq "realfull") {
+    if ($action eq "full" || $action eq "realfull" || $action eq "baremetal" ) {
         if ($line =~ /name SECTION: Module configuration options/) {
             $done = 1;
         }
 
         if (!$done && $line =~ m!^//\s?#define! &&
-                ( $line !~ /$exclude_re/ || $line =~ /$no_exclude_re/ ) ) {
+                ( $line !~ /$exclude_re/ || $line =~ /$no_exclude_re/ ) &&
+                ( $action ne "baremetal" || ( $line !~ /$exclude_baremetal_re/ ) ) ) {
```

Comment:
This condition looks really complex to me. I wonder if it could be made simpler by appending `@excluded_baremetal` to `@excluded` above near the place where you do that for `@non_excluded`? Or am I missing something?

Reply:
I thought so in the begining but it can't be done because of wildcard nature of both exclude and not exclude lists. 
```non_excluded``` adds back all platform ```_ALT```s. In order to remove things for baremetal a second pass is needed. That's why I only merged ```@non_excluded``` and ```@non_excluded_baremetal```.